nixos/modules/system/boot/luksroot.nix

```diff
@@ -19,6 +19,18 @@ let
         echo "ok"
     fi
 
+    ${optionalString (header != null) ''
+    if ! test -e ${header}; then
+        echo -n "waiting 10 seconds for header ${header} to appear..."
```

Comment:
Why do you need to poll for this? Shouldn't the file be there after `udevadm settle`?